src/docfx/restore/Restore.cs

```diff
@@ -23,52 +23,51 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
             using (Progress.Start("Restore dependencies"))
             {
                 var restoredDocsets = new ConcurrentDictionary<string, int>(PathUtility.PathComparer);
-                restoredDocsets.TryAdd(docsetPath, 0);
 
-                // Root docset must have a config
-                var (configErrors, config) = Config.Load(docsetPath, options, extend: false);
-                ReportErrors(report, configErrors);
-                report.Configure(docsetPath, config);
+                await RestoreDocset(docsetPath, true);
 
-                await RestoreOneDocset(report, docsetPath, options, config, RestoreDocset, restoreLocRepo: true, @implicit);
-
-                async Task RestoreDocset(string docset)
+                async Task RestoreDocset(string docset, bool root)
                 {
-                    if (restoredDocsets.TryAdd(docset, 0) && Config.LoadIfExists(docset, options, out var loadErrors, out var childConfig, false))
+                    if (restoredDocsets.TryAdd(docset, 0))
                     {
+                        var (errors, config) = Config.TryLoad(docset, options, extend: false);
+                        ReportErrors(report, errors);
+
+                        if (root)
+                        {
+                            report.Configure(docsetPath, config);
+                        }
+
                         // no need to restore child docsets' loc repository
-                        await RestoreOneDocset(report, docset, options, childConfig, RestoreDocset, restoreLocRepo: false, @implicit);
+                        await RestoreOneDocset(docset, config, async subDocset => await RestoreDocset(subDocset, false), restoreLocRepo: root);
                     }
                 }
             }
-        }
 
-        private static async Task RestoreOneDocset(
-            Report report,
-            string docsetPath,
-            CommandLineOptions options,
+            async Task RestoreOneDocset(
+            string docset,
```

Comment:
>string docset, [](start = 12, length = 14)

minor: indent it. Personally prefer to keep it a private method as the logic is not quite simple.